### ISSUE TITLE
`line.start_with?("ssh")` is not a good test...

### ISSUE THREAD

**Issue by stuart-shelton-hp** [nickola/chef-ssh-keys#14]
... since valid keys can begin with "ssh", or "ecdsa", or "ed25519" - and likely more types in the future.

Additionally, why be selective at all - de-duplicating is fair enough, but if the user has additional entries or comments in their existing file, why not simply maintain these and add any new keys if they don't already exist?

All the best,

Stuart

**Comment by glensc** [comment]
yup, it totally breaks keys which have some extra attributes, like:

```
$ less ~/.ssh/authorized_keys
from="192.168.2.1,127.0.0.1",no-agent-forwarding,no-port-forwarding,no-x11-forwarding,no-user-rc ssh-dss AAAAB3...
```
